packages/fxa-settings/src/pages/ResetPassword/ResetPasswordConfirmed/container.tsx

```diff
@@ -63,35 +71,56 @@ const ResetPasswordConfirmedContainer = ({
   };
 
   const getOauthRedirect = async () => {
-    const { error, redirect } = await finishOAuthFlowHandler(
+    return await finishOAuthFlowHandler(
       uid,
       sessionToken,
       keyFetchToken,
       unwrapBKey
     );
-
-    return { redirect, error };
   };
 
-  const continueWithVerifiedSession = async () => {
+  const validateOauthFlow = async (result: FinishWithOAuthResult) => {
     if (isOAuthIntegration(integration)) {
-      const { redirect, error } = await getOauthRedirect();
+      const { error, redirect, code, state } = await getOauthRedirect();
       if (error) {
         handleOAuthRedirectError(error);
         return;
       }
+
+      result({ error, redirect, code, state });
+    }
+  };
+
+  const continueWithVerifiedSession = async () => {
+    validateOauthFlow(({ redirect }) => {
       if (redirect) {
         hardNavigate(redirect);
         return;
       }
-    }
+    });
     navigateWithQuery(`/settings`);
   };
 
   if (oAuthDataError) {
     return <OAuthDataError error={oAuthDataError} />;
   }
 
+  validateOauthFlow(({ error, redirect, code, state }) => {
+    // To keep the type-system happy; error is already handled.
+    if (error) {
+      return;
+    }
+
+    if (integration.isSync()) {
```

Comment:
I think maybe we actually don't need these changes in `ResetPasswordConfirmed` 🤔 The path to this is `reset_password_verified`, and the only place I see in the code that we take users there is `if (isOAuthIntegration(integration)) {`, and mobile/oauth desktop will only be truthy with `if (isOAuthNativeIntegration(integration))`.

Reply:
Closing this loop: we moved the code back to `CompleteResetPassword` and added the `isSync` check in there too after the oauth integration check.